modules/secops-rules/main.tf

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  reference_lists = try(yamldecode(file(var.factories_config.reference_lists)), var.reference_lists_config)
+  reference_lists_entries = {
+    for k, v in local.reference_lists : k => split("\n", file("${var.factories_config.reference_lists_defs}/${k}.txt"))
+  }
+  reference_list_type_mapping = {
+    STRING = "REFERENCE_LIST_SYNTAX_TYPE_PLAIN_TEXT_STRING"
+    REGEX  = "REFERENCE_LIST_SYNTAX_TYPE_REGEX"
+    CIDR   = "REFERENCE_LIST_SYNTAX_TYPE_CIDR"
+  }
+  secops_rules = {
+    for file_name in fileset(var.factories_config.rules_defs, "*.yaral") : replace(file_name, ".yaral", "") => file("${var.factories_config.rules_defs}/${file_name}")
+  }
+  secops_rule_deployment = try(yamldecode(file(var.factories_config.rules)), var.rules_config)
+}
+
+resource "google_chronicle_reference_list" "reference_list" {
+  for_each          = local.reference_lists
+  project           = var.project_id
+  location          = var.tenant_config.region
+  instance          = var.tenant_config.customer_id
+  reference_list_id = each.key
+  description       = each.value.description
+  dynamic "entries" {
+    for_each = toset(local.reference_lists_entries[each.key])
```

Comment:
toset is not needed here and can introduce ordering issues with the API response